pkg/cmd/server/api/v1/types.go

```diff
@@ -99,6 +99,11 @@ type MasterConfig struct {
 	// PauseControllers instructs the master to not automatically start controllers, but instead
 	// to wait until a notification to the server is received before launching them.
 	PauseControllers bool `json:"pauseControllers,omitempty"`
+	// ControllerLeaseTTL enables controller election, instructing the master to attempt to acquire
+	// a lease before controllers start and renewing it within a number of seconds defined by this value.
+	// Setting this value non-negative forces pauseControllers=true. This value defaults off (0, or
+	// omitted) and controller election can be disabled with -1.
+	ControllerLeaseTTL int `json:"controllerLeaseTTL,omitempty"`
```

Comment:
It's omitempty, so it doesn't add anything.  Hrm... shouldn't be omit empty.

On Mon, Aug 17, 2015 at 10:04 AM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/cmd/server/api/v1/types.go
> <https://github.com/openshift/origin/pull/4137#discussion_r37190116>:
>
> > @@ -99,6 +99,11 @@ type MasterConfig struct {
> >  	// PauseControllers instructs the master to not automatically start controllers, but instead
> >  	// to wait until a notification to the server is received before launching them.
> >  	PauseControllers bool `json:"pauseControllers,omitempty"`
> > +	// ControllerLeaseTTL enables controller election, instructing the master to attempt to acquire
> > +	// a lease before controllers start and renewing it within a number of seconds defined by this value.
> > +	// Setting this value non-negative forces pauseControllers=true. This value defaults off (0, or
> > +	// omitted) and controller election can be disabled with -1.
> > +	ControllerLeaseTTL int `json:"controllerLeaseTTL,omitempty"`
>
> omitempty :-/ On Aug 17, 2015, at 10:01 AM, David Eads <
> notifications@github.com> wrote: In pkg/cmd/server/api/v1/types.go <
> https://github.com/openshift/origin/pull/4137#discussion_r37189810>:
> @@ -99,6 +99,11 @@ type MasterConfig struct { // PauseControllers
> instructs the master to not automatically start controllers, but instead //
> to wait until a notification to the server is received before launching
> them. PauseControllers bool `json:"pauseControllers,omitempty"` + //
> ControllerLeaseTTL enables controller election, instructing the master to
> attempt to acquire + // a lease before controllers start and renewing it
> within a number of seconds defined by this value. + // Setting this value
> non-negative forces pauseControllers=true. This value defaults off (0, or +
> // omitted) and controller election can be disabled with -1. +
> ControllerLeaseTTL int `json:"controllerLeaseTTL,omitempty"`
> This should have required a test update in
> https://github.com/openshift/origin/blob/master/pkg/cmd/server/api/v1/types_test.go.
> — Reply to this email directly or view it on GitHub <
> https://github.com/openshift/origin/pull/4137/files#r37189810>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4137/files#r37190116>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift